diff -ru a/DEFSPTOOL.BAS b/DEFSPTOOL.BAS
--- a/DEFSPTOOL.BAS	2024-06-29 18:09:42.397054968 +0900
+++ b/DEFSPTOOL.BAS	2024-06-29 18:10:57.937055937 +0900
@@ -2,7 +2,7 @@
    20 /* sprite pattern editor ver.1.2
    30 /*   Copyright (C) 1987,1989 SHARP
   100 /******initialize********
-  110 str help_file="\ETC\DEFSPTOOL.HLP"
+  110 str help_file="DEFSPTOOL.HLP"
   120 dim int  pal(15,15) /*palet data*/
   130 dim char w(255)     /*pattern work*/
   140 int  l=1            /*lut no*/
@@ -169,13 +169,13 @@
 14050   cls
 14060   color 2
 14070   print "        パレットの設定"
-14080   while (count < 200)
+14080   while (count < 2000000)
 14090     mspos(px,py)
 14100     sp_move(0,px,py)
 14110     msstat(x,y,bl,br)
 14120     if (bl) then break else count=count+1
 14130   endwhile
-14140   if (count < 200 and px >= X3 and px < X4 and py >= Y33 and py < Y34 ) then {
+14140   if (count < 2000000 and px >= X3 and px < X4 and py >= Y33 and py < Y34 ) then {
 14150     sp_set(0,,,253)
 14160     c_old=c
 14170     c=rgb(r,g,b)+i
@@ -209,13 +209,13 @@
 15060   fill(X3,Y35+n*16,X4-1,Y35+n*16+11,62)
 15070   set_func(n)
 15080   set_msg(n)
-15090   while(i<200)
+15090   while(i<2000000)
 15100     mspos(px,py)
 15110     sp_move(0,px,py)
 15120     msstat(x,y,bl,br)
 15130     if (bl) then break else i=i+1
 15140   endwhile
-15150   if (i < 200 and px >= X3 and px < X4 and py >= Y35+n*16 and py < Y35+n*16+11) then {
+15150   if (i < 2000000 and px >= X3 and px < X4 and py >= Y35+n*16 and py < Y35+n*16+11) then {
 15160     fill(X3,Y35+n*16,X4-1,Y35+n*16+11,1984)
 15170     set_func(n)
 15180     if n < 8 then {
@@ -504,13 +504,13 @@
 28030   fill(X3,Y37,X4-1,Y38-1,62)
 28040   set_func(16)
 28050   set_msg(16)
-28060   while(i < 200)
+28060   while(i < 2000000)
 28070     mspos(px,py)
 28080     sp_move(0,px,py)
 28090     msstat(x,y,bl,br)
 28100     if (bl) then break else i=i+1
 28110   endwhile
-28120   if (i < 200 and px >= X3 and px < X4 and py >= Y37 and py < Y38) then {
+28120   if (i < 2000000 and px >= X3 and px < X4 and py >= Y37 and py < Y38) then {
 28130     fill(X3,Y37,X4-1,Y38-1,1984)
 28140     set_func(16)
 28150     console 0,31,1
@@ -754,7 +754,7 @@
 37570       a=itoa(lno)+"     "
 37580       for k=0 to 15
 37590         n=d(j*16+k)
-37600         a=a+"&H"+hex$(n)+","
+37600         a=a+right$(itoa(n),2)+","
 37610       next
 37620       if (j <> 15) then {
 37630         a=a+chr$(13)+chr$(10)
@@ -843,11 +843,11 @@
 39000 /*----------------------
 39010 func ferr(mode)
 39020   int i=0,x,y,bl,br
-39030   dim str msg(2)[34]
-39040   msg(0) = "ファイルのオープンに"+chr$(13)+chr$(10)+"失敗しました"
-39050   msg(1) = "ファイルの書き込みに"+chr$(13)+chr$(10)+"失敗しました"
-39070   msg(2) = "パターン番号が無効です"
-39080   /*          
+39030   dim str msg(2)[34]={
+39040            "ファイルのオープンに失敗しました",
+39050            "ファイルの書き込みに失敗しました",
+39070            "パターン番号が無効です"
+39080           }
 39090   cls
 39100   locate 0,22,0
 39110   color 9
