diff --git a/libgloss/read.c b/libgloss/read.c
index 83a3b549b..867648278 100644
--- a/libgloss/read.c
+++ b/libgloss/read.c
@@ -20,7 +20,7 @@ extern char inbyte (void);
  * read  -- read bytes from the serial port. Ignore fd, since
  *          we only have stdin.
  */
-int
+_READ_WRITE_RETURN_TYPE
 read (int fd,
        void *buf_,
        size_t nbytes)
diff --git a/libgloss/write.c b/libgloss/write.c
index 80bcd4a77..757141291 100644
--- a/libgloss/write.c
+++ b/libgloss/write.c
@@ -21,7 +21,7 @@ extern int  outbyte (char x);
  *          stdout and stderr are the same. Since we have no filesystem,
  *          open will only return an error.
  */
-int
+_READ_WRITE_RETURN_TYPE
 write (int fd,
        const void *buf_,
        size_t nbytes)
